builtin/providers/google/resource_container_node_pool.go

```diff
@@ -0,0 +1,148 @@
+package google
+
+import (
+	"log"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/container/v1"
+	"google.golang.org/api/googleapi"
+)
+
+func resourceContainerNodePool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceContainerNodePoolCreate,
+		Read:   resourceContainerNodePoolRead,
+		Delete: resourceContainerNodePoolDelete,
```

Comment:
Done. Can you help me now figure out whether there's anything that can be taken out of the other functions now that the Exists one is there?